src/ts/signers.ts

```diff
@@ -16,13 +16,16 @@ import {
  * Takes a Signer instance on creation.
  * All other Signer methods are proxied to initial instance.
  */
-export class TypedDataV3Signer implements TypedDataSigner {
+export class TypedDataVersionedSigner implements TypedDataSigner {
   signer: Signer;
   provider: JsonRpcProvider;
   _isSigner = true;
+  _signMethod: string;
 
-  constructor(signer: Signer) {
+  constructor(signer: Signer, version?: "v1" | "v2" | "v3" | "v4") {
```

Comment:
nit:
```suggestion
  constructor(signer: Signer, version?: "v3" | "v4") {
```

v1 and v2 have different JSON RPC interfaces, so in order to support them you would need to change the payload that you are building. Specifically, I think the signing message blob isn't stringified for v1 and v2 (there are some other minor differences as well).

I don't think its worth the effort at the moment to implement, so I would just disallow it.